modules/openapi-generator/src/main/resources/dart-next/serialization/repository_base.mustache

```diff
@@ -0,0 +1,60 @@
+import 'dart:async';
+
+/// The general rule for implementing this class is
+/// json == serialize<T>(deserialize<T>(json))
+/// object == deserialize<T>(serialize<T>(object))
+/// Where json in this context means a dart primitive
+/// JSON = String | num | bool | null | Uint8List | List<Json> | Map<String, Json>
+abstract class SerializationRepositoryBase {
+    const SerializationRepositoryBase();
+    
+    FutureOr<Object?> serialize<T extends Object?>(T src, TypeInfo inputTypeInfo, {Object? context,});
+    FutureOr<T> deserialize<T extends Object?>(Object? value, TypeInfo targetTypeInfo, {Object? context,});
```

Comment:
Making these async seems overkill and increases complexity in other places.
What is your use case here?

Reply:
@kuhnroyal you might want to offset the (de)serialization process to another isolate like the dart generator does

Reply:
I see. Does it make sense to provide both options so that we can use the sync access especially for parameter serialization etc.?

Reply:
@kuhnroyal that was actually my plan since I sometimes need the serialization logic in places where I can't use async